access-functions.php

```diff
@@ -619,6 +619,23 @@ function get_graphql_setting( string $option_name, $default = '', $section_name
 	return apply_filters( 'graphql_get_setting_section_field_value', $value, $default, $option_name, $section_fields, $section_name );
 }
 
+/**
+ * Return the GraphQL Endpoint for the site.
+ *
+ * @return string
+ */
+function graphql_get_endpoint() {
+
+	$graphql_endpoint = trailingslashit( site_url() ) . get_graphql_setting( 'graphql_endpoint', 'graphql' );
+
+	/**
+	 * Filter the endpoint before returning
+	 *
+	 * @param string $graphql_endpoint The graphql endpoint for the site
+	 */
+	return apply_filters( 'graphql_get_endpoint', $graphql_endpoint );
+}
```

Comment:
Should this maybe be a wrapper for a method in the `Router` class (and possibly called `graphql_get_endpoint_url()` instead)?

We've got similar functionality [in `Router::init()`](https://github.com/wp-graphql/wp-graphql/blob/57302c08ab655ac0720263c42272318e493a45af/src/Router.php#L60), as well as a similarly named filter that actually alters the endpoint used by WPGraphQL instead of just what gets output on the header. 

Thinking maybe a single source of truth and/or more clarity in the method/filter names would be helpful to dx.

Reply:
@justlevine pushed an update.

The `Router::$route` deals with the endpoint route specifically, not the full url. I need the full url to be consistent here.

Reply:
Perhaps the new filter is unnecessary 🤔

Reply:
GH isnt showing me the new commit yet.

What I _blindly_ want to suggest is to instead create a new method (either in Utils or a top-level access function) something like:

```php
public static function get_graphql_endpoint() : string {
  $filtered_endpoint = apply_filters( 'graphql_endpoint', null );
  return  filtered_endpoint ? $filtered_endpoint : get_graphql_setting( 'graphql_endpoint', 'graphql' );
}
```
then use that in both [`Router::init()`](https://github.com/wp-graphql/wp-graphql/blob/0e6618a5da8282ab443c5ddebbc0d86fdd99730c/src/Router.php#L60), [`Setting::register_settings()`,](https://github.com/wp-graphql/wp-graphql/blob/0e6618a5da8282ab443c5ddebbc0d86fdd99730c/src/Admin/Settings/Settings.php#L95) _and_ our new `get_graphql_endpoint_url()` (which probably doesn't need a filter), and _then_ use that to populate our `$headers` which could either be filtered with the existing [`graphql_response_headers_to_send` filter](https://github.com/wp-graphql/wp-graphql/blob/0e6618a5da8282ab443c5ddebbc0d86fdd99730c/src/Router.php#L377), or a new filter if we need to modify headers [before they get sent to the query analyzer](https://github.com/wp-graphql/wp-graphql/blob/0e6618a5da8282ab443c5ddebbc0d86fdd99730c/src/Router.php#L353).

Reply:
ah, ya. makes sense.